src/apps/euler/ethereum/euler.e-token.token-fetcher.ts

```diff
@@ -85,47 +84,42 @@ export class EthereumEulerETokenTokenFetcher implements PositionFetcher<AppToken
 
         if (totalSupply.isZero() || !underlyingToken) return null;
 
-        const pricePerShare = Number(totalSupply.toString()) / Number(market.totalBalances);
+        const pricePerShare = await eTokenContract.convertBalanceToUnderlying(utils.parseEther('1'));
 
         const dataProps = {
-          name: `Euler E token ${market.name}`,
+          name: market.name,
           liquidity: Number(totalSupply) * underlyingToken.price,
           underlyingAddress: market.id,
           interestRate: Number(market.interestRate) / 10 ** 18,
-          borrowAPY: Number(market.borrowAPY) / 10 ** 18,
-          supplyAPY: Number(market.borrowAPY) / 10 ** 18,
+          supplyAPY: Number(market.supplyAPY) * 100 / 1e27,
           totalSupply: totalSupply.toString(),
           totalBalances: market.totalBalances,
         };
 
         return {
           address: market.eTokenAddress,
           symbol: `E${market.symbol}`,
-          name: `Euler E token ${market.name}`,
+          name: `${market.name} (E)`,
           type: ContractType.APP_TOKEN as const,
-          supply: Number(market.totalSupply) / 10 ** Number(market.decimals),
-          pricePerShare,
+          supply: Number(market.totalBalances) / 10 ** Number(market.decimals),
+          pricePerShare: underlyingToken.price * Number(utils.formatEther(pricePerShare)),
```

Comment:
Should be `1` (pricePerShare is the ratio between the token and the underlying token price)

See https://studio.zapper.fi/docs/concepts/app-tokens#what-is-pricepershare-why-is-it-useful